sharedbuffers/mapped_struct.py

```diff
@@ -943,6 +943,8 @@ def _eq(self, other):
     def __len__(self):
         if self.elem_step == 0:
             return self._metadata()[1]
+        elif self.elem_start == self.elem_end:
+            return 0
         elif self.elem_step < 0:
             return (self.elem_start - self.elem_end - 1) / (-self.elem_step) + 1
```

Comment:
What if `start < end`?

Or the converse for step > 0.

You should probably normalize that in the constructor:

``` python
if step < 0:
    end = min(start, end)
elif step > 0:
    start = min(start, end)
```

Then you don't need to worry about the in other places

Reply:
That's a good idea, yes.